internal/exec/engine.go

```diff
@@ -189,28 +191,49 @@ func (e Engine) fetchProviderConfig(f resource.Fetcher) (types.Config, error) {
 // set, the referenced and evaluted config will be returned. Otherwise, if
 // "ignition.config.append" is set, each of the referenced configs will be
 // evaluated and appended to the provided config. If neither option is set, the
-// provided config will be returned unmodified.
-func (e *Engine) renderConfig(cfg types.Config, f resource.Fetcher) (types.Config, error) {
+// provided config will be returned unmodified. An updated fetcher will be
+// returned with any new timeouts set.
+func (e *Engine) renderConfig(cfg types.Config, f resource.Fetcher) (types.Config, resource.Fetcher, error) {
 	// Apply any new timeout info before fetching other configs.
 	e.client = resource.NewHttpClient(e.Logger, cfg.Ignition.Timeouts)
 	if cfgRef := cfg.Ignition.Config.Replace; cfgRef != nil {
-		return e.fetchReferencedConfig(*cfgRef, f)
+		newCfg, err := e.fetchReferencedConfig(*cfgRef, f)
+
+		// Replace the HTTP client in the fetcher to be configured with the
+		// timeouts of the new config
+		client := resource.NewHttpClient(e.Logger, newCfg.Ignition.Timeouts)
+		f.Client = &client
+
+		newCfg, f, err = e.renderConfig(newCfg, f)
+		if err != nil {
+			return newCfg, f, err
```

Comment:
shouldn't we return an empty `types.Config`, nil, and the err here? As the code currently reads this if statement has no effect.

Reply:
This code here actually looks comical now that I'm looking back at it. I'll just `return e.renderConfig(...`

Reply:
I guess I've got `if err != nil {` hammered into my head too well.